src/lib/ibcclient.ts

```diff
@@ -731,7 +746,15 @@ export class IbcClient {
         },
       }),
     };
-    this.logger.debug(`MsgUpdateClient`, updateMsg);
+
+    this.logger.debug(
+      `MsgUpdateClient`,
+      formatMessage(updateMsg, (mutableMsg, encode) => {
+        if (mutableMsg.value.header?.value) {
+          mutableMsg.value.header.value = encode(mutableMsg.value.header.value);
```

Comment:
`s/encode/toBase64/` here would let us remove the `encode` argument above and make this more flexible

Reply:
I used `encode` argument because we could extend it with length check and use hex for short arrays as you suggested.

I agree being explicit is better (more readable), also the function itself is generic now: bf99247